src/dash-table/derived/cell/wrapperStyles.ts

```diff
@@ -16,28 +18,47 @@ const partialGetter = (
 ) => traverseMap2(
     data,
     columns,
-    (datum, column, i) => getDataCellStyle(datum, i + offset.rows, column)(styles)
+    (datum, column, i) => getDataCellStyle(datum, i + offset.rows, column, false, false)(styles)
 );
 
 const getter = (
-    styles: CSSProperties[][],
+    baseline: CSSProperties[][],
+    columns: Columns,
+    styles: IConvertedStyle[],
+    data: Data,
     offset: IViewportOffset,
+    activeCell: ICellCoordinates,
     selectedCells: SelectedCells
 ) => {
-    styles = shallowClone(styles);
+    baseline = shallowClone(baseline);
+
+    const cells = selectedCells.length ?
+        selectedCells :
+        activeCell ? [activeCell] : [];
+
+    const activeStyles = styles.filter(style => style.checksActive());
+    const inactiveStyles = styles.filter(style => !style.checksActive());
 
     R.forEach(({ row: i, column: j }) => {
-        i -= offset.rows;
-        j -= offset.columns;
+        const iNoOffset = i - offset.rows;
+        const jNoOffset = j - offset.columns;
 
-        if (i < 0 || j < 0 || styles.length <= i || styles[i].length <= j) {
+        if (iNoOffset < 0 || jNoOffset < 0 || baseline.length <= iNoOffset || baseline[iNoOffset].length <= jNoOffset) {
             return;
         }
 
-        styles[i][j] = R.merge(styles[i][j], SELECTED_CELL_STYLE);
-    }, selectedCells);
+        const active = isActiveCell(activeCell, i, j);
+
+        const style = {
+            ...getDataCellStyle(data[i], i + offset.rows, columns[j], active, true)(inactiveStyles),
+            ...SELECTED_CELL_STYLE,
+            ...getDataCellStyle(data[i], i + offset.rows, columns[j], active, true)(activeStyles)
```

Comment:
Make an exception for active/selected, apply !active/!selected styles first, apply the baseline, then apply active/selected styles